framework/email/tasks.py

```diff
@@ -4,50 +4,84 @@
 
 from framework.tasks import app
 from website import settings
+import sendgrid
 
 logger = logging.getLogger(__name__)
 
 
 @app.task
 def send_email(from_addr, to_addr, subject, message, mimetype='html', ttls=True, login=True,
-                username=None, password=None, mail_server=None):
+                username=None, password=None, mail_server=None, categories=None):
     """Send email to specified destination.
     Email is sent from the email specified in FROM_EMAIL settings in the
     settings module.
 
+    Uses the Sendgrid API if ``settings.SENDGRID_API_KEY`` is set.
+
     :param from_addr: A string, the sender email
     :param to_addr: A string, the recipient
     :param subject: subject of email
     :param message: body of message
+    :param tuple categories: Categories to add to the email using SendGrid's
+        SMTPAPI. Used for email analytics.
+        See https://sendgrid.com/docs/User_Guide/Statistics/categories.html
+        This parameter is only respected if using the Sendgrid API.
+        ``settings.SENDGRID_API_KEY`` must be set.
 
     :return: True if successful
     """
-    username = username or settings.MAIL_USERNAME
-    password = password or settings.MAIL_PASSWORD
-    mail_server = mail_server or settings.MAIL_SERVER
-
     if not settings.USE_EMAIL:
         return
-    if login and (username is None or password is None):
-        logger.error('Mail username and password not set; skipping send.')
-        return
+    if settings.SENDGRID_API_KEY:
+        return send_with_sendgrid(
+            from_addr=from_addr,
+            to_addr=to_addr,
+            subject=subject,
+            message=message,
+            mimetype=mimetype,
+            categories=categories
+        )
+    else:
```

Comment:
I'd prefer to see this branch factored out into a function as well. Maybe `send_with_smtp`?